src/clf/lr_training.py

```diff
@@ -0,0 +1,182 @@
+"""
+Description:
+    Training a logistic regression model for predicting probabilities of medical specialties
+"""
+from array import array
+import pandas as pd
+import numpy as np
+import pickle
+import imblearn
+import os
+
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV
+
+from sklearn.pipeline import Pipeline
+from imblearn.pipeline import Pipeline as imbPipeline
+
+from traitlets import List
+
+from utils import load_data
+from utils import preprocessing_pipeline
+
+# Build pipeline
+def build_pipeline(pipeline) -> imblearn.pipeline.Pipeline:
+    """
+    Build pipeline for model
+
+    Parameters
+    ----------
+    preprocessing_pipeline : imblearn.pipeline.Pipeline
+        pipeline with preprocessing steps
+    Returns
+    -------
+    imblearn.pipeline.Pipeline
+        pipeline for model
+    """
+    pipeline.steps.append(
+        (
+            "clf",
+            LogisticRegression(
+                random_state=42,
+                multi_class="multinomial",
+                penalty="l1",
+                solver="saga",
+            ),
+        )
+    )
+    return pipeline
+
+
+# Fit model
+def fit_model(
+    model: imblearn.pipeline.Pipeline,
+    X_train: pd.core.series.Series,
+    y_train: pd.core.series.Series,
+) -> imblearn.pipeline.Pipeline:
+    """
+    Fit model
+
+    Parameters
+    ----------
+    model : imblearn.pipeline.Pipeline
+        pipeline for model
+    X_train : pd.core.series.Series
+        train data
+    y_train : pd.core.series.Series
+        train labels
+
+    Returns
+    -------
+    imblearn.pipeline.Pipeline
+        fitted model
+    """
+    model.fit(X_train, y_train)
+    return model
```

Comment:
Im Zweifel kannst du diese Methode auch rausnehmen da sie nur eine Zeile lang ist. Da ist es schneller sie direkt in der main methode auszuführen